### PR TITLE
Qa 1278 - update env file key and minor refactoring

### PR DESCRIPTION
**Pull request by suneel-lean**
## Description of Changes

1. Updated the environment files
2. Updated the s3 service file
3. Added the .gitignore file
4. Added CODEOWNERS
5. Updated IaC topics.csv with repo codeowner details in this [pr](https://github.com/leantechnologies/IaC/pull/1035)
6. Added linter and pre-commit hook

## Types of Changes

* [ ] Bug fix
* [ ] New feature
* [ ] Performance improvement
* [ ] Documentation update
* [x] Refactoring

## Testing
NA

## Additional Notes

1. Checked the linter for Dockerfile using hadolint and there are no errors, only warning and info is provided:
image

## Checklist

* [ ] I have performed sufficient testing.
* [ ] I have updated the documentation accordingly (if applicable).
* [ ] I have considered potential breaking changes or compatibility issues.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,111 @@
...
+
+# Compiled binary addons (https://nodejs.org/api/addons.html)
+build/Release
+
+# Dependency directories
+node_modules/
+jspm_packages/
+
+# Result directories 
+allure-report/
```
Comment (ReshailLean) [inline .gitignore:48]: We dont have allure report in this service

File: services/s3Service.ts [hunk 2]
```diff
@@ -4,19 +4,21 @@ import { S3 } from 'aws-sdk';
...
-var bucketName = process.env.BUCKETNAME
-var namespace = process.env.NAMESPACE
-var region = process.env.REGION
-// https://axjj8th4b6iy.compat.objectstorage.me-jeddah-1.oraclecloud.com
-var accessKeyId = process.env.ACCESSKEY
-var secretAccessKey = process.env.SECRETKEY
+var bucketName = process.env.HTTP_LOGGING_BUCKET_NAME
+var region = process.env.HTTP_LOGGING_BUCKET_REGION
+var s3_endpoint = HTTP_LOGGING_BUCKET_ENDPOINT
+if (s3_endpoint == undefined || s3_endpoint == null) {
```
Comment (ReshailLean) [inline services/s3Service.ts:10]: if you did !s3_endpoint would cover both these cases.

**Review by ReshailLean — APPROVED** [review]

File: services/s3Service.ts (hunk 2 above)
Comment (suneel-lean) [reply services/s3Service.ts]: corrected

File: .gitignore (hunk 1 above)
Comment (suneel-lean) [reply .gitignore]: corrected

**Review by ReshailLean — APPROVED** [review]

**Pull request merged**